### PR TITLE
Add dependency on UI if providers are compiled with gui

### PR DESCRIPTION
**Pull request by m-kuhn**
Fixes failing builds if ui files are not processed before building providers.

```
In file included from /builddir/build/BUILD/qgis-3.9.0/src/providers/wms/qgswmsprovidergui.cpp:18:
/builddir/build/BUILD/qgis-3.9.0/src/providers/wms/qgswmssourceselect.h:20:10: fatal error: ui_qgswmssourceselectbase.h: No such file or directory
   20 | #include "ui_qgswmssourceselectbase.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

CC @neteler @daniviga

### CODE REVIEW

**Review by daniviga — CHANGES_REQUESTED** [review]
I get a CMake error on this branch:

```
CMake Error at src/providers/grass/CMakeLists.txt:189 (ADD_DEPENDENCIES):
  Cannot add target-level dependencies to non-existent target
  "grassprovider".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.
Call Stack (most recent call first):
  src/providers/grass/7/CMakeLists.txt:7 (ADD_GRASSLIB)
```

File: src/providers/grass/CMakeLists.txt [hunk 1]
```diff
@@ -186,6 +186,7 @@ MACRO(ADD_GRASSLIB GRASS_BUILD_VERSION)
       TARGET_LINK_LIBRARIES(grassprovider${GRASS_BUILD_VERSION}
         qgis_gui
       )
+      ADD_DEPENDENCIES(grassprovider ui)
```
Comment (daniviga) [inline src/providers/grass/CMakeLists.txt]:
```suggestion

```

Comment (m-kuhn) [reply src/providers/grass/CMakeLists.txt]: Sorry, it should be

```suggestion
      ADD_DEPENDENCIES(grassprovider${GRASS_BUILD_VERSION} ui)
```

Can you confirm?

Comment (daniviga) [reply src/providers/grass/CMakeLists.txt]: I confirm that it works with this last fix. LGTM

**Review by daniviga — APPROVED** [review]

**Review by nyalldawson — APPROVED** [review]

**Comment by landryb** [comment]
Fwiw, i'm still seeing the same build failure as reported in the first comment, building 3.10 release on OpenBSD. 
```
In file included from /usr/obj/ports/qgis-3.10.0/qgis-3.10.0/src/providers/wms/qgswmsdataitems.cpp:26:
/usr/obj/ports/qgis-3.10.0/qgis-3.10.0/src/providers/wms/qgswmssourceselect.h:20:10: fatal error: 'ui_qgswmssourceselectbase.h' file not found
#include "ui_qgswmssourceselectbase.h"

```
Could this be some fallout/side-effect from https://github.com/qgis/qgis/commit/635c1b54d8025ef2c02e5fecf1537ad5ac77f4d7#diff-17c63f5441b19b50cf0d99055932fb4c ? It's as if a dependency wasnt correctly declared somewhere... but only on the wms provider.

**Pull request merged**